website/addons/citations/provider.py

```diff
@@ -17,6 +17,17 @@ def __init__(self, provider_name):
     def serializer(self):
         pass
 
+    def check_credentials(self, node_addon):
+        valid = True
+        try:
+            node_addon.api.client
```

Comment:
However, client is a cached property, i.e. it will try to connect to Mendeley the first time and will error if it can't connect. The second time `client` is accessed, a request will *not* be sent to Mendeley, and an error will not be raised.

This is a bug: the first call to `check_credentials` with invalid creds will return `False`. The second call, however, will always return `True`. It's probably best to instantiate the Mendeley client here rather than rely on the `client` property.